ppl/src/main/java/com/amazon/opendistroforelasticsearch/sql/ppl/plans/logical/HasInput.java

```diff
@@ -0,0 +1,20 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.ppl.plans.logical;
+
+public interface HasInput<T, R> {
```

Comment:
Please add JavaDoc for generic type too.

Reply:
Ok will do, thanks!

Reply:
could we make the node Immutable by using the context to maintain the current parent command?